src/main/java/org/folio/auth/authtokenmodule/MainVerticle.java

```diff
@@ -629,16 +645,36 @@ we generate a custom token (since we have that power) that
     logger.debug("Getting user permissions for " + username + " (userId " +
             userId + ")");
     long startTime = System.currentTimeMillis();
-    Future<PermissionData> retrievedPermissionsFuture = usePermissionsSource
-      .getUserAndExpandedPermissions(userId, tenant, okapiUrl, permissionsRequestToken,
-        requestId, extraPermissions);
+
+    // Need to check if the user is still active
+    Future<PermissionData> retrievedPermissionsFuture;
+    if (finalUserId != null && !finalUserId.isEmpty()) {
```

Comment:
What is happening here? Empty or null? SQ says one.. IMHO, get rid of the isEmpty.

Reply:
Check to make sure user id is not null and is not empty (length = 0). Does that make sense?